core/src/main/java/org/elasticsearch/common/xcontent/support/AbstractXContentParser.java

```diff
@@ -92,13 +95,27 @@ public boolean isBooleanValue() throws IOException {
 
     @Override
     public boolean booleanValue() throws IOException {
+        boolean interpretedAsLenient = false;
+        boolean booleanValue;
+        String rawValue = null;
+
         Token token = currentToken();
         if (token == Token.VALUE_NUMBER) {
-            return intValue() != 0;
+            interpretedAsLenient = true;
+            booleanValue = intValue() != 0;
+            rawValue = String.valueOf(intValue());
         } else if (token == Token.VALUE_STRING) {
-            return Booleans.parseBoolean(textCharacters(), textOffset(), textLength(), false /* irrelevant */);
+            rawValue = new String(textCharacters(), textOffset(), textLength());
```

Comment:
We can only specify a `Charset` when constructing a `String` from a `byte[]` but not when constructing from a `char[]`. The reason is that in Java a `char` always represents a [Unicode scalar value](http://www.unicode.org/reports/tr27/#notation) whereas a `byte` does not (necessarily).